daemon/src/maker_cfd.rs

```diff
@@ -347,6 +238,96 @@ where
     }
 }
 
+#[xtra_productivity]
+impl<O, M, T, W> Actor<O, M, T, W> {
+    async fn handle_accept_rollover(&mut self, msg: AcceptRollOver) -> Result<()> {
+        if self
+            .rollover_actors
+            .send(&msg.order_id, rollover_maker::AcceptRollOver)
+            .await
+            .is_err()
+        {
+            tracing::warn!(%msg.order_id, "No active rollover");
+        }
+
+        Ok(())
+    }
+}
+
+impl<O, M, T, W> Actor<O, M, T, W>
+where
+    O: xtra::Handler<oracle::GetAnnouncement> + xtra::Handler<oracle::MonitorAttestation>,
+    M: xtra::Handler<monitor::StartMonitoring>,
+    T: xtra::Handler<maker_inc_connections::TakerMessage>
+        + xtra::Handler<Stopping<rollover_maker::Actor>>
+        + xtra::Handler<RollOverProposed>,
+    W: 'static,
+    Self: xtra::Handler<Stopping<rollover_maker::Actor>>,
+{
+    async fn handle_propose_roll_over(
+        &mut self,
+        proposal: RollOverProposal,
+        taker_id: Identity,
+        ctx: &mut Context<Self>,
+    ) -> Result<()> {
+        tracing::info!(
+            "Received proposal from the taker {}: {:?} to roll over order {}",
+            taker_id,
+            proposal,
+            proposal.order_id
+        );
+
+        // check if CFD is in open state, otherwise we should not proceed
+        let mut conn = self.db.acquire().await?;
+        let cfd = load_cfd_by_order_id(proposal.order_id, &mut conn).await?;
+        match cfd {
+            Cfd {
+                state: CfdState::Open { .. },
+                ..
+            } => (),
+            _ => {
+                anyhow::bail!("Order is in invalid state. Cannot propose roll over.")
+            }
+        };
+
+        let this = ctx.address().expect("acquired own address");
+
+        let (rollover_actor_addr, _rollover_actor_future) = rollover_maker::Actor::new(
```

Comment:
You can't just ignore this future otherwise the actor will not run :upside_down_face:

Add it to `self.tasks`.